app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -294,6 +298,9 @@ async function addEthereumChainHandler(
         fromNetworkConfiguration: getProviderConfig(),
       },
     });
+    if (hasPermission(req.origin)) {
+      setNetworkClientIdForDomain(req.origin, networkConfigurationId);
```

Comment:
I wonder if we can also add this directly in the `SelectedNetworkController` by listening for accepted `switchEthereumChain` requests from the `ApprovalController`? wdyt @Gudahtt @mcmire @BelfordZ

Reply:
Looked into this doesn't look very straightforward unfortunately.